packages/blade/src/components/IconButton/StyledIconButton.native.tsx

```diff
@@ -0,0 +1,55 @@
+import type { ReactElement } from 'react';
+import { useState } from 'react';
+import styled from 'styled-components/native';
+
+import type { StyledIconButtonProps } from './StyledIconButton.d';
+import { makeAccessible } from '~utils';
+import type { ColorContrastTypes } from '~tokens/theme/theme';
+
+type GetIconColorTokenParams = {
+  isPressed: boolean;
+  contrast: ColorContrastTypes;
+};
+
+type State = 'active' | 'default';
+type IconColorToken = `feedback.neutral.action.icon.link.${State}.${ColorContrastTypes}Contrast`;
+
+const getIconColorToken = ({ isPressed, contrast }: GetIconColorTokenParams): IconColorToken => {
+  const contrastType = contrast === 'high' ? 'highContrast' : 'lowContrast';
+  const state = isPressed ? 'active' : 'default';
+
+  return `feedback.neutral.action.icon.link.${state}.${contrastType}`;
+};
+
+type StyledPressableProps = {
+  contrast: ColorContrastTypes;
+};
+
+const StyledPressable = styled.Pressable<StyledPressableProps>({
+  alignSelf: 'flex-start', // ensure button only takes needed width
+});
+
+const StyledIconButton = ({
+  icon: Icon,
+  onClick,
+  size,
+  contrast,
+  accessibilityLabel,
+}: StyledIconButtonProps): ReactElement => {
+  const [isPressed, setIsPressed] = useState(false);
+  const iconColorToken = getIconColorToken({ isPressed, contrast });
+
+  return (
+    <StyledPressable
```

Comment:
The clickable area is too small for a thumb to be able to record a press. Should add a [`hitSlop`](https://reactnative.dev/docs/pressable#hitslop) here?

Reply:
[Similar thoughts](https://razorpay.slack.com/archives/G01B3LQ9H0W/p1660135022356289) but I think we want to wait for some research on this before taking a call.

Reply:
Responded there. We do need some hitSlop at least for IconButton. This would be too small for a human thumb to be able to press it

Reply:
Holding on to this so we've design inputs first. Think we can merge this in for the moment and revisit later with more design inputs.

Reply:
Tracking https://github.com/razorpay/blade/issues/639